runtime/databricks/automl_runtime/forecast/pmdarima/model.py

```diff
@@ -0,0 +1,323 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from typing import List
+
+import pandas as pd
+import mlflow
+import pmdarima
+from mlflow.exceptions import MlflowException
+from mlflow.protos.databricks_pb2 import INVALID_PARAMETER_VALUE
+
+
+class ArimaModel(mlflow.pyfunc.PythonModel):
+    """
+    ARIMA mlflow model wrapper for univariate forecasting.
+    """
+
+    def __init__(self, pickled_model: bytes, horizon: int, frequency: str,
+                 start_ds: pd.Timestamp, end_ds: pd.Timestamp, time_col: str) -> None:
+        """
+        Initialize the mlflow Python model wrapper for ARIMA.
+        :param pickled_model: the pickled ARIMA model as a bytes object.
+        :param horizon: int number of periods to forecast forward.
+        :param frequency: the frequency of the time series
+        :param start_ds: the start time of training data
+        :param end_ds: the end time of training data
+        :param time_col: the column name of the time column
+        """
+        self._pickled_model = pickled_model
+        self._horizon = horizon
+        self._frequency = frequency
+        self._start_ds = start_ds
+        self._end_ds = end_ds
+        self._time_col = time_col
+        super().__init__()
+
+    def load_context(self, context: mlflow.pyfunc.model.PythonModelContext) -> None:
+        """
+        Loads artifacts from the specified PythonModelContext that can be used by
+        PythonModel.predict when evaluating inputs. When loading an MLflow model with
+        load_pyfunc, this method is called as soon as the PythonModel is
+        constructed.
+        :param context: A PythonModelContext instance containing artifacts that the model
+                        can use to perform inference.
+        """
+        from pmdarima.arima import ARIMA  # noqa: F401
+        return
+
+    def model(self) -> pmdarima.arima.ARIMA:
+        """
+        Deserialize the ARIMA model by pickle.
+        :return: ARIMA model
+        """
+        import pickle
+        return pickle.loads(self._pickled_model)
+
+    def predict_timeseries(self, horizon: int = None) -> pd.DataFrame:
+        """
+        Predict target column for given horizon and history data.
+        :param horizon: int number of periods to forecast forward.
+        :return: A pd.DataFrame with the forecasts and confidence intervals for given horizon and history data.
+        """
+        horizon = horizon or self._horizon
+        future_pd = self._forecast(horizon)
+        in_sample_pd = self._predict_in_sample()
+        return pd.concat([in_sample_pd, future_pd]).reset_index(drop=True)
+
+    def predict(self, context: mlflow.pyfunc.model.PythonModelContext, model_input: pd.DataFrame) -> pd.Series:
+        """
+        Predict API from mlflow.pyfunc.PythonModel. Returns the prediction values for given timestamps in the input
```

Comment:
Nit: prefer to have a one-line summary first, followed by an empty line, followed by detailed documentation. https://www.python.org/dev/peps/pep-0257/#multi-line-docstrings

ditto for other docstrings.